From 4d52a04813d76807b42d62379f828aa35b941abf Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Fri, 1 Aug 2014 15:21:14 +0800
Subject: [PATCH] usb/dwc3-device: fix kernel panic in
 dwc3_endpoint_transfer_complete()

There is a possibility that dep->endpoint.desc is modified to NULL
by another thread, after the spinlock in dwc3_endpoint_transfer_complete()
is released for a while.

So need to check this pointer before it's accessed.

Change-Id: I7c6a7381ccec8cf9c16a2190ae2eee8bf5301088
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
(cherry picked from commit 8565c6a27954ac94e98c9841744c4338d5fdc288)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/usb/dwc3/gadget.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index d266697..50908cf 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -1894,7 +1894,7 @@ static int dwc3_cleanup_done_reqs(struct dwc3 *dwc, struct dwc3_ep *dep,
 			break;
 	} while (1);
 
-	if (usb_endpoint_xfer_isoc(dep->endpoint.desc) &&
+	if (dep->endpoint.desc && usb_endpoint_xfer_isoc(dep->endpoint.desc) &&
 			list_empty(&dep->req_queued)) {
 		if (list_empty(&dep->request_list)) {
 			/*
-- 
1.7.9.5

